iOS/SceneCoordinator.swift

```diff
@@ -723,7 +723,30 @@ class SceneCoordinator: NSObject, UndoableCommandRunner, UnreadCountProvider {
 		}
 		
 		currentFeedIndexPath = indexPath
-		masterFeedViewController.updateFeedSelection(animations: animations)
+		
+		// I know this is a bad pyramid of doom - would like to know how to best improve this
+		if let ip = currentFeedIndexPath, let node = nodeFor(ip) {
```

Comment:
You could change this into a single `if` or `guard` and format the conditions so that they are stacked to prevent all the indenting.